test/README.md

```diff
@@ -0,0 +1,67 @@
+Candid test data
+================
+
+This directory contains Candid test data, to be used to check Candid
+implementations for compliance.
+
+This test suite covers
+
+ * parsing Candid service descriptions
+ * given a Candid type, decoding binary Candid data, asserting success or failure
+ * given a Candid type, decoding textual Candid data, asserting success or failure
+
+Because this test data necessarily has to be host-language agnostic, it is not
+possible to
+
+ * assert the decoded value, as it would be host-language specific
```

Comment:
But we have a text representation for values. So can't we at least assert the equality of a binary value against an equivalent text value?

Reply:
It might also be useful to compare different representations of the same value. I'd suggest a general assertion like
```
<argseq> <input> `==` <input>
```
Perhaps also useful:
```
<argseq> <input> `!=` <input>
```

Reply:
> But we have a text representation for values. So can't we at least assert the equality of a binary value against an equivalent text value?

Can you give a concrete example?

Do you mean an assertion like `decode  "DIDL\00\00" at type () yields ()`? But that requires the test suite to not only decode candid binary into host langauge values, but also serialize host language values into Candid textual representation. Which is not what we want to test here, and not something we would require of each implementation (do we?)?

For example, Motoko’s RTS currently does not support parsing or printing textual representation.

Reply:
Comparision is interesting! This would allow us to assert, for example, that subtyping ignores a certain field…

I assume the `==` here is the host language’s equality on decoded values. I guess that is a general enough concept to except to be available in most languages – even Motoko(!), once #1812 is done.

Reply:
> Can you give a concrete example?

I meant something like that comparison.

> I assume the == here is the host language’s equality on decoded values

I think we can directly define an abstract syntax of Candid values, and structural equality on those -- if we couldn't, then Candid wouldn't be very meaningful :).

Reply:
Concretely, with that you can say
```
(Int) blob "..." = 42
```

Reply:
Oh, right.

It will still mean “decoding the binary `DIDL\00\7d\*` yields the same host value as parsing the textual Candid `(42)`”
and not “decoding the binary `DIDL\00\7d\*` yields the host value 42”, but close enough.